model/user.go

```diff
@@ -884,24 +885,38 @@ func UsersWithGroupsAndCountFromJson(data io.Reader) *UsersWithGroupsAndCount {
 	return uwg
 }
 
-var passwordRandomSource = rand.NewSource(time.Now().Unix())
+type lockedRand struct {
+	mu sync.Mutex
+	rn *rand.Rand
+}
+
+func (r *lockedRand) Intn(n int) int {
+	r.mu.Lock()
+	m := r.rn.Intn(n)
+	r.mu.Unlock()
+	return m
+}
+
+var passwordRandom = lockedRand{
+	rn: rand.New(rand.NewSource(time.Now().Unix())),
+	mu: sync.Mutex{},
```

Comment:
I don't think this is necessary since it's a value. Same for the one in the test case below.